File: python/pascals_triangle_II.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1, 1]
        row = [1]
        last_row = self.getRow(rowIndex - 1)
        for i in range(len(last_row) - 1):
            row.append(last_row[i] + last_row[i+1])
        row.append(1)
        return row

File: C++/pascals_triangle_II.cpp
//no. 119
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        if (rowIndex == 0){
            return vector<int> {1};
        }else if(rowIndex == 1){
            return vector<int> {1, 1};
        }
        vector<int> row = {1};
        vector<int> lastRow = getRow(rowIndex - 1);
        for (int i = 0;i < lastRow.size() - 1; i++){
            row.push_back(lastRow[i] + lastRow[i+1]);
        }
        row.push_back(1);
        return row;
    }
};
